packages/api-client/src/libs/scalarScripts.ts

```diff
@@ -0,0 +1,74 @@
+import type { ResponseTest } from '@scalar/oas-utils/entities/workspace/spec'
+import type { AxiosRequestConfig, AxiosResponse } from 'axios'
+
+function getPreRequestScalarObject(config: AxiosRequestConfig) {
+  const _config = config
+  return {
+    environment: {
+      set: (key: string, value: string) => {},
+      get: (key: string) => '',
+    },
+    request: {
+      headers: {
+        add({ key, value }: { key: string; value: string }) {
+          _config.headers = _config.headers || {}
+          _config.headers[key] = value
+        },
+      },
+    },
+    _config,
+  }
+}
+export const executePreRequestScript = (
+  script: string,
+  config: AxiosRequestConfig,
+) => {
+  const scalar = getPreRequestScalarObject(config)
+  const executeScript = new Function('scalar', script)
+
+  const output = executeScript(scalar)
+
+  return { output, config: scalar._config }
+}
+
+function expect(actual: any) {
+  return {
+    toBe: (expected: any) => {
```

Comment:
Would be nice to have user generated messages here. Should take an optional second argument that is  `(actual: any, expected: any): string`

Reply:
If its provided then we should run that function to generate the message and fallback to our message